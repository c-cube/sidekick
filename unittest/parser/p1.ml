module P = Parse_term
module A = Ast_term

(*
let () = Printexc.record_backtrace true
*)

let () =
  Printexc.register_printer (function
    | Parser_comb.ParseError e -> Some (Parser_comb.Error.to_string e)
    | _ -> None)

let test_str what s =
  let t = P.of_string_exn s in

  Fmt.printf "%s: %a@." what A.pp_term t;
  Fmt.printf "loc(%s): %a@." what A.pp_loc (A.loc t)

let () = test_str "t1" "f (g x) y"
let () = test_str "t2" "let x:= 1 in f (f x 2)"
