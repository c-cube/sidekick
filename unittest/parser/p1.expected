t1: f (g x) y
loc(t1): at line 0, column 0 - at line 0, column 9
t2: let x := 1 in (f (f x 2))
loc(t2): at line 0, column 0 - at line 0, column 21
